tensorboard/webapp/widgets/range_input/range_input_component.ts

```diff
@@ -345,14 +345,21 @@ export class RangeInputComponent implements OnInit, OnDestroy {
     // range  [100, 500]  → delete upper value → single [100, X]
     // single [100, X]    → delete lower value → single [50, X]
     // range  [100, 500]  → delete lower value → range  [50, 500]
+    // single [100, X]    → add    upper value → range  [100, 500]
     if (this.isUpperValueRemoved(input, position)) {
+      this.upperValue = null;
+      this.singleValueChanged.emit(this.lowerValue);
```

Comment:
This introduces composite actions - two actions being fired in quick succession. Can you model the underlying actions differently?

Reply:
Thanks for pointing this out. That's a great point. Here I then removed the output `upperValueRemoved`.
The output is originally designed for "toggle useRange" and cause `useRangeSelectTime` to change. But then I realized `useRangeSelectTime` is set based on `upperValue !== null`. so on removing upper value, all I need to do is emit the `singleValueChanged` (which set `selectedTme.end` to null, which means upperValue is null and sets `useRangeSelectTime` to false)

To sum,
- upperValueRemoved output is removed
- `this.upperValue = null` is removed

PTAL and let me know what do you think. Thanks!